### ISSUE TITLE
Layer list highligts the wrong selected layer after certain undo/redo operations

### ISSUE THREAD

**Issue by Estecka** [JannisX11/blockbench#2582]
### What are you trying to do, and what do you expect to happen?

After disabling layers on a texture, certain strings of undo/redo that would restore them will cause those layers to become seemingly impossible to select. This is only a visual bug; the layers *are* selected, but the highlight does not update to reflect that.

This only affects layers that were restored by an undo operation. Layers on other textures work properly. After manually re-enabling layers on the affected texture (creating a whole new set of layers), the new layers are not affected.

This cannot be fixed by selecting another texture, then reselecting the affected texture.

This can be fixed by closing and re-opening the project.

### Repro:
1. Create a new minecraft skin. (The nature of the model doesn't seem to matter.)
2. Enable layers on the texture
3. Create a duplicate of the layer.
4. Disable layers on the texture.
5. Hit Undo (restores layers), then Redo (disables layers), then Undo again (restores layers). Hitting Undo only once is not enough to reproduce the issue.
6. Try to select a different layer: **The intended layer will not be highlighted when clicking on it.**
7. Try painting or using colour adjustment tools: **The tools take effect not on the highlighted layer, but on the layer you previously intended to select.**
8. Try toggling the visibility of a layer: **The highlight will be updated and moved to the layer you previously intended to select.**

### What happens instead?

**Expected behaviour:** The selected layer should always be highlighted.

**Observed behaviour:** The layer list no longer updates the highlight untila layer's visibility is toggled.

### Model format in which the issue occurs

Minecraft Skin & Optifine Model

### Blockbench variant

Program

### Blockbench version

4.11.2

### Operating System

Windows 10

### Installed Blockbench plugins

None
